packages/firebase/index.d.ts

```diff
@@ -5434,6 +5434,88 @@ declare namespace firebase.storage {
      *     including if the object did not exist.
      */
     updateMetadata(metadata: firebase.storage.SettableMetadata): Promise<any>;
+    /**
+     * List all items (files) and prefixes (folders) under this storage reference.
+     *
+     * It is a helper method for calling list() recursively until there is no
+     * more results. The default pagination size is 200.
+     *
+     * Note: the results may not be a consistent snapshot if objects are changed
+     * between paginating requests.
+     *
+     * Warning: If there are
+     *
+     * @return A promise that resolves with all the items and prefixes under
+     *      the current storage reference. `prefixes` contains references to
+     *      sub-directories and `items` contains references to objects in this
+     *      folder. `nextPageToken` is never returned.
+     */
+    listAll(): Promise<ListResult>;
+    /**
+     * List items (files) and prefixes (folders) under this storage reference.
+     *
+     * GCS is a key-blob store. Firebase storage impose the semantic of '/'
+     * delimited folder structure.
+     * Objects whose paths, aside from the prefix, contain '/' will have
+     * their path, truncated after the '/', returned in prefixes.
+     * Duplicate prefixes are omitted.
+     * Objects whose paths, aside from the prefix, do not contain '/' will
+     * be returned in items.
+     *
+     * Firebase storage does not support invalid object paths that end with
```

Comment:
s/Firebase storage/Cloud Storage for Firebase (?) or "Firebase Storage"

I would also remove "invalid" since they are not invalid in GCS.